src/pages/index.js

```diff
@@ -111,19 +174,88 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      //submitBtn.textContent = "Save";
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails. 


Please, reuse `setButtonText` because it contains the default button texts


```js
    .finally(() => {
      setButtonText(submitBtn, false);
    });
```